### PR TITLE
leddar_one: split into separate header, implementation, and main

### PR DESCRIPTION
**Pull request by dagar**
- split into separate header, implementation, and main
 - remove default serial port (doesn't make any sense to be set in the driver)
 - update copyright year
 - cleanup old headers

### CODE REVIEW

File: src/drivers/distance_sensor/leddar_one/LeddarOne.cpp [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+ ****************************************************************************/
+
+#include "LeddarOne.hpp"
+
+#include <stdlib.h>
+#include <string.h>
+
+LeddarOne::LeddarOne(const char *serial_port, uint8_t device_orientation):
+	ScheduledWorkItem(MODULE_NAME, px4::serial_port_to_wq(serial_port)),
+	_px4_rangefinder(0 /* device id not yet used */, ORB_PRIO_DEFAULT, device_orientation)
```
Comment (TSC21) [inline src/drivers/distance_sensor/leddar_one/LeddarOne.cpp]: @dagar would this be a good opportunity to make the device ID usable?

Comment (dagar) [reply src/drivers/distance_sensor/leddar_one/LeddarOne.cpp]: All that's really missing is to add them here. https://github.com/PX4/Firmware/blob/master/src/drivers/drv_sensor.h

I was going to get it cleaned up a bit to try and have a dedicated range for the existing sensors, but it doesn't really matter.

Comment (TSC21) [reply src/drivers/distance_sensor/leddar_one/LeddarOne.cpp]: What if you have multiple sensors of the same type? How do you distinguish their ID's? And how do you match the ID with the uORB instance you want to use?

Comment (dagar) [reply src/drivers/distance_sensor/leddar_one/LeddarOne.cpp]: The device ID contains the type (drv_sensor.h), bus, and address. The orb message contains the device id.

There's a helper to decode them that's used in listener. For example `listener sensor_accel` on fmu-v4 shows 2 accels with type 0x37 (DRV_ACC_DEVTYPE_ICM20602) and 0x16 (DRV_ACC_DEVTYPE_MPU9250).

Screen Shot 2019-12-31 at 10 44 34 AM

**Pull request merged**